### PR TITLE
docs: remove empty onInit() & constructors

### LINKED ISSUE #48575: OnInit is not added to new components any more
### Describe the problem that you experienced

The linked stackoverflow topic do explain the issue well!
> `ng generate component` does not create ngOnInit and constructor

### Enter the URL of the topic with the problem

https://stackoverflow.com/q/74870515/42580

### Describe what you were looking for in the documentation

I was trying to run the TourOfHeros-tutorial when I could not match my experiance with the tutorial, so after some google-work I found the SO-topic.

### Describe the actions that led you to experience the problem

_No response_

### Describe what you want to experience that would fix the problem

_No response_

### Add a screenshot if that helps illustrate the problem

_No response_

### If this problem caused an exception or error, please paste it here

_No response_

### If the problem is browser-specific, please specify the device, OS, browser, and version

_No response_

### Provide any additional information here in as much as detail as you can

_No response_

### PR DESCRIPTION
**Pull request by JeanMeche**
Following  angular/angular-cli#24008, the schematics don't produce ngOnInit and the constructor anymore. Let's reflect that in the Tour of Heroes tutorial.

Edit: I also removed it from other parts of the documentation. 

Fixes #48575

## PR Type
What kind of change does this PR introduce?

- [x] Documentation content changes

### CODE REVIEW

**Review by hexiaokang — APPROVED** [review]

**Review by dylhunn — APPROVED** [review]
Looks like https://github.com/angular/angular/issues/24008 is unrelated. Did you mean to link to a different PR?

reviewed-for: fw-core, fw-forms

**Comment by JeanMeche** [comment]
My bad, It's pointing at the wrong repo : angular/angular-cli#24008,

File: packages/forms/src/directives/checkbox_value_accessor.ts [hunk 1]
```diff
@@ -48,10 +48,16 @@ export const CHECKBOX_VALUE_ACCESSOR: any = {
...
    */
   writeValue(value: any): void {
-    this.setProperty('checked', value);
+    if (value === 'indeterminate') {
+      this.setProperty('indeterminate', true);
+      this.setProperty('checked', false);
+    } else {
+      this.setProperty('indeterminate', false);
+      this.setProperty('checked', value);
+    }
```
Comment (AndrewKushnir) [inline packages/forms/src/directives/checkbox_value_accessor.ts:61]: @JeanMeche it looks like this change is unrelated, could you please revert it?

**Review by AndrewKushnir — APPROVED** [review]
LGTM 👍

(this PR will require a rebase once PR https://github.com/angular/angular/pull/48546 lands, adding the "cleanup" label for now)

**Comment by AndrewKushnir** [comment]
@JeanMeche PR https://github.com/angular/angular/pull/48546 is now merged, could you please rebase this PR and resolve merge conflicts?

**Comment by JeanMeche** [comment]
@AndrewKushnir ✅

**Review by AndrewKushnir — APPROVED** [review]

**Comment by AndrewKushnir** [comment]
**Caretaker note**: low risk PR, current set of approvals should be sufficient.

**Review by jessicajaniuk — APPROVED** [review]
LGTM

**Comment by alxhub** [comment]
This PR was merged into the repository by commit 43ced455d687c13ade2c5f61bf1ededad098c05a.
